conclave-host/src/main/kotlin/com/r3/conclave/host/internal/GramineEnclaveHandle.kt

```diff
@@ -38,41 +41,39 @@ class GramineEnclaveHandle(
         enclaveInterface = SocketHostEnclaveInterface(maxConcurrentCalls)
     }
 
-    companion object {
-        const val GRAMINE_ENCLAVE_JAR_NAME = "enclave-shadow.jar"
-        const val GRAMINE_ENCLAVE_MANIFEST = "java.manifest"
-
-        private val logger = loggerFor<GramineEnclaveHandle>()
-    }
-
     override fun initialise() {
-        try {
-            /** Bind a port for the interface to use. */
-            val port = enclaveInterface.bindPort()
-
-            /**
-             * Start the enclave process, passing the port that the call interface is listening on.
-             * TODO: Implement a *secure* method for passing port to the enclave.
-             */
-            processGramineDirect = ProcessBuilder()
-                .inheritIO()
-                .directory(enclaveDirectory.toFile())
-                .command("gramine-direct", "java", "-cp", GRAMINE_ENCLAVE_JAR_NAME, "com.r3.conclave.enclave.internal.GramineEntryPoint", port.toString())
-                .start()
-
-            /** Wait for the local call interface start process to complete. */
-            enclaveInterface.start()
-
-            /** Initialise the enclave. */
-            enclaveInterface.initializeEnclave(enclaveClassName)
-        } catch (t: Throwable) {
-            this.destroy()
-            throw t
+        /** Bind a port for the interface to use. */
+        val port = enclaveInterface.bindPort()
+
+        /**
+         * Start the enclave process, passing the port that the call interface is listening on.
+         * TODO: Implement a *secure* method for passing port to the enclave.
+         */
+        val command = mutableListOf(
+            "gramine-direct",
+            "java", "-cp", GRAMINE_ENCLAVE_JAR, "com.r3.conclave.enclave.internal.GramineEntryPoint", port.toString()
+        )
+        if ((workingDirectory / PYTHON_FILE).exists()) {
+            command += PYTHON_FILE
         }
+
+        processGramineDirect = ProcessBuilder()
+            .inheritIO()
+            .directory(workingDirectory.toFile())
+            .command(command)
+            .start()
+
+        // Avoid leaving an orphan child process
+        Runtime.getRuntime().addShutdownHook(Thread(processGramineDirect::destroyForcibly))
```

Comment:
I have noticed orphaned enclave processes on my dev machine, but it's hard to say what caused them. They could have been caused by my forgetting to call `EnclaveHost.close` in the integration tests. Regardless, this is a good safety net in case things fall through.